client/dashboard/utils/site-plans.ts

```diff
@@ -31,19 +31,23 @@ export const isSitePlanBigSkyTrial = ( site: Site ) => {
 };
 
 export const isSitePlanPaid = ( site: Site ) => {
-	if ( ! site.plan ) {
-		return false;
-	}
-
 	return ! [ DotcomPlans.JETPACK_FREE, DotcomPlans.FREE_PLAN ].includes(
-		site.plan.product_slug as DotcomPlans
+		site.plan?.product_slug as DotcomPlans
 	);
 };
 
 export const isSitePlanHostingTrial = ( site: Site ) => {
-	if ( ! site.plan ) {
-		return false;
-	}
+	return site.plan?.product_slug === DotcomPlans.HOSTING_TRIAL_MONTHLY;
+};
+
+export const isSiteOnECommerceTrial = ( site: Site ) => {
+	return site.plan?.product_slug === DotcomPlans.ECOMMERCE_TRIAL_MONTHLY;
+};
+
+export const isSiteOnMigrationTrial = ( site: Site ) => {
+	return site.plan?.product_slug === DotcomPlans.MIGRATION_TRIAL_MONTHLY;
+};
 
-	return site.plan.product_slug === DotcomPlans.HOSTING_TRIAL_MONTHLY;
+export const isSiteLaunchable = ( site: Site ) => {
```

Comment:
I would expect a function called `isSiteLaunchable` to also check whether the site has already been launched, etc. This function seems specifically about plans.

Maybe it could be called

```suggestion
export const isLaunchableSitePlan = ( site: Site ) => {
```

Reply:
And since the predicate is so specialised, maybe it shouldn't be promoted to utils? Just inline it in `<TrialUpsellNotice>` like they do in the old Calypso code?

Reply:
Renamed to isSitePlanLaunchable to be consistent with others like isSitePlanPaid. I think having them in utils promotes shared usage, and avoid different places having different implementations. Since most of the code in site-plans.ts are only used for site visibility (for now), I'm happy to move it inline in the meantime. Updated in 5969f7f.